catalog/app/containers/Admin/UsersAndRoles/Users.tsx

```diff
@@ -0,0 +1,1070 @@
+import cx from 'classnames'
+import * as FF from 'final-form'
+import invariant from 'invariant'
+import * as R from 'ramda'
+import * as React from 'react'
+import * as redux from 'react-redux'
+import * as RF from 'react-final-form'
+import * as M from '@material-ui/core'
+import * as Sentry from '@sentry/react'
+
+import * as Pagination from 'components/Pagination'
+import * as Notifications from 'containers/Notifications'
+import * as Auth from 'containers/Auth'
+import * as Dialogs from 'utils/GlobalDialogs'
+import * as GQL from 'utils/GraphQL'
+import assertNever from 'utils/assertNever'
+import * as Format from 'utils/format'
+import * as validators from 'utils/validators'
+
+import * as Form from '../Form'
+import * as Table from '../Table'
+import * as RoleSelect from './RoleSelect'
+
+import USERS_QUERY from './gql/Users.generated'
+import USER_CREATE_MUTATION from './gql/UserCreate.generated'
+import USER_DELETE_MUTATION from './gql/UserDelete.generated'
+import USER_SET_EMAIL_MUTATION from './gql/UserSetEmail.generated'
+import USER_SET_ROLE_MUTATION from './gql/UserSetRole.generated'
+import USER_SET_ACTIVE_MUTATION from './gql/UserSetActive.generated'
+import USER_SET_ADMIN_MUTATION from './gql/UserSetAdmin.generated'
+
+import { UserSelectionFragment as User } from './gql/UserSelection.generated'
+
+type Role = GQL.DataForDoc<typeof USERS_QUERY>['roles'][number]
+
+const DIALOG_PROPS: Dialogs.ExtraDialogProps = { maxWidth: 'xs', fullWidth: true }
+
+const useDialogFormStyles = M.makeStyles((t) => ({
+  root: {
+    marginTop: t.spacing(-3),
```

Comment:
I think "(-2)" is nicer. So, I think, the padding for title and dialog content is ok. The only problem is top margin for the first text field, and it's a 16px, not 24px.
Otherwise, the margin under the dialog title ("Invite user") and under the section title ("Assign roles") looks the same. But, dialog title margin seems intuitively should be bigger.

Reply:
i see the logic behind this, but -3 _feels_ better. well, whatever